tests/test_algorithm_performance.py

```
@@ -0,0 +1,418 @@
+        assert benchmark.duration < 0.1, "Cache cleanup affecting performance"
+
+    def test_diversity_calculation_performance(self):
+        """Test diversity calculation performance for semantic operators."""
+        ideas = self.create_test_ideas(20)
+        
+        # Test simple content-based diversity calculation
+        with PerformanceBenchmark("diversity-simple") as benchmark:
+            # Simulate Jaccard similarity calculation
+            diversity_scores = []
+            for i, idea1 in enumerate(ideas):
+                for j, idea2 in enumerate(ideas[i+1:], i+1):
```

Ruff (0.12.2)

Loop control variable `j` not used within loop body

Rename unused `j` to `_j`

(B007)